code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/filter/ServiceAddRemoveLinkServiceValidationFilter.java

```diff
@@ -0,0 +1,59 @@
+package io.cattle.platform.servicediscovery.api.filter;
+
+import io.cattle.platform.core.model.Service;
+import io.cattle.platform.iaas.api.filter.common.AbstractDefaultResourceManagerFilter;
+import io.cattle.platform.servicediscovery.api.constants.ServiceDiscoveryConstants;
+import io.cattle.platform.servicediscovery.api.dao.ServiceConsumeMapDao;
+import io.cattle.platform.util.type.CollectionUtils;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+import io.github.ibuildthecloud.gdapi.request.resource.ResourceManager;
+import io.github.ibuildthecloud.gdapi.validation.ValidationErrorCodes;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+public class ServiceAddRemoveLinkServiceValidationFilter extends AbstractDefaultResourceManagerFilter {
+
+    @Inject
+    ServiceConsumeMapDao consumeMapDao;
+
+    private static final Map<String, Boolean> ACTIONS;
+
+    static {
+        ACTIONS = new HashMap<>();
```

Comment:
I agree that it is a bit of an overkill in terms of the amount of code.  But I do feel it'd be a lot more clear.  The original use of the map felt non-intuitive.  I think I'd even prefer just having a Set with 2 constants and doing the comparisons explicitly.